locationdelegation/src/main/java/com/google/androidbrowserhelper/locationdelegation/LocationDelegationExtraCommandHandler.java

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.androidbrowserhelper.locationdelegation;
+
+import android.content.Context;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.browser.trusted.TrustedWebActivityCallbackRemote;
+
+import com.google.androidbrowserhelper.trusted.ExtraCommandHandler;
+
+public class LocationDelegationExtraCommandHandler implements ExtraCommandHandler {
+    static final String CHECK_LOCATION_PERMISSION_COMMAND_NAME = "checkAndroidLocationPermission";
+    private static final String START_LOCATION_COMMAND_NAME = "startLocation";
+    private static final String STOP_LOCATION_COMMAND_NAME = "stopLocation";
+
+    private boolean mIsProviderGmsCore;
```

Comment:
Instead of this, could you have a `private final LocationListener` and set it in the constructor (checking if the Play Services are available there)?

Or if the Play Services check could take some time, we could have a `getLocationListener()` method that initializes it lazily.

(Basically so that we don't have an `if (mIsProviderGmsCore)` in both `startLocationProvider` and `stopLocationProvider`.)